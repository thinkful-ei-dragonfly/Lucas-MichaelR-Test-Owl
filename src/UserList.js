import React from 'react';

import User from './User';
import { userInfo } from 'os';

// filter STORE.participants.inSession === true

// stage will filter STORE.participants.inSession === true and STORE.participants.onStage === true

function UserList(props) {

  // const participants = props.participants;
  const participants = props.participants;
  let users = participants.sort(function(a, b) {
    return b.onStage - a.onStage;
  }).map(user => <User key={user.id} display={'inList'}avatar={user.avatar} name={user.name} onStage={user.onStage}/>);

  return <div className="UserList">
    <p>This will be a list of users</p>
    {users}
  </div>
}

export default UserList;