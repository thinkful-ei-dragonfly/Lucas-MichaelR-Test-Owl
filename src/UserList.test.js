import React from 'react';
import renderer from 'react-test-renderer';

import UserList from './UserList';

import STORE from './store';

describe('<UserList  />', () => {
  it('renders without crashing [snapshot test]', () => {
    // Render this component as JSON
    const tree = renderer.create(<UserList participants={STORE.participants}/>).toJSON();
    // Checks wether this matches the previous snapshot
    // The previous snapshot will be stored in __snapshots__/UserList .test.js
    expect(tree).toMatchSnapshot();
  })
})