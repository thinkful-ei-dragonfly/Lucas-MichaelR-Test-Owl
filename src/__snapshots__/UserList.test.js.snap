// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList  /> renders without crashing [snapshot test] 1`] = `
<div
  className="UserList"
>
  <div
    className="User"
  >
    <div
      className="image-wrapper"
    >
      <img
        alt="user_avatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="user-info"
    >
      <h4>
        Koren Templeton
      </h4>
      <p
        className="user-status"
      >
        on stage
      </p>
    </div>
  </div>
  <div
    className="User"
  >
    <div
      className="image-wrapper"
    >
      <img
        alt="user_avatar"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="user-info"
    >
      <h4>
        Ashla Attwool
      </h4>
      <p
        className="user-status"
      >
        on stage
      </p>
    </div>
  </div>
  <div
    className="User"
  >
    <div
      className="image-wrapper"
    >
      <img
        alt="user_avatar"
        src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="user-info"
    >
      <h4>
        Caty Flucker
      </h4>
      <p
        className="user-status"
      >
        in session
      </p>
    </div>
  </div>
  <div
    className="User"
  >
    <div
      className="image-wrapper"
    >
      <img
        alt="user_avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="user-info"
    >
      <h4>
        Axe Kubicka
      </h4>
      <p
        className="user-status"
      >
        in session
      </p>
    </div>
  </div>
  <div
    className="User"
  >
    <div
      className="image-wrapper"
    >
      <img
        alt="user_avatar"
        src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="user-info"
    >
      <h4>
        Frank Runciman
      </h4>
      <p
        className="user-status"
      >
        in session
      </p>
    </div>
  </div>
</div>
`;
